ldm/models/diffusion/ddim.py

```diff
@@ -14,17 +14,25 @@
 
 
 class DDIMSampler(object):
-    def __init__(self, model, schedule='linear', device='cuda', **kwargs):
+    def __init__(self, model, schedule='linear', device='', **kwargs):
         super().__init__()
         self.model = model
         self.ddpm_num_timesteps = model.num_timesteps
         self.schedule = schedule
-        self.device = device
+        if device == '':
+            if torch.cuda.is_available():
+                self.device = 'cuda'
+            elif torch.backends.mps.is_available():
+                self.device = 'mps'
+            else:
+                self.device = 'cpu'
+        else:
+            self.device = device
```

Comment:
The [`T2I._set_sampler`](https://github.com/lstein/stable-diffusion/blob/main/ldm/simplet2i.py#L570) method already takes care of this by setting the device arg for all samplers to the `T2I.device`. So all of these hardcoded set devices appear to be superfluous to me. Moreover, if the `T2I.device` was set to `cuda:1` instead of the default `cuda:0` this code might break image generation by moving the sampler tensors to the incorrect device.